File: play-handler-classes/src/main/kotlin/org/example/demo/demos/MethodAnnoNotTakeEffectKotlinDemo.java
package org.example.demo.demos;

import com.alibaba.csp.sentinel.annotation.SentinelResource;

import static org.example.demo.biz.Greeting.doGreeting;

@SuppressWarnings("unused")
public abstract class MethodAnnoNotTakeEffectKotlinDemo {
    @SentinelResource(
            blockHandlerClass = {BlockHandlerClass.class},
            fallbackClass = {FallbackClass.class}
    )
    public String greeting1(String name) {
        return doGreeting(name);
    }

    @SentinelResource(
            blockHandlerClass = BlockHandlerClass.class,
            fallbackClass = FallbackClass.class
    )
    public String greeting2(String name) {
        return doGreeting(name);
    }

    @SentinelResource(
            blockHandlerClass = {BlockHandlerClass.class, BlockHandlerClass2.class},
            fallbackClass = {FallbackClass.class, FallbackClass2.class}
    )
    public String greeting3(String name) {
        return doGreeting(name);
    }
}

File: play-handler-classes/src/main/kotlin/org/example/hc/demos/MethodAnnoNotTakeEffectKotlinDemo.kt
package org.example.hc.demos

import com.alibaba.csp.sentinel.annotation.SentinelResource
import org.example.hc.biz.Greeting

@Suppress("unused")
abstract class MethodAnnoNotTakeEffectKotlinDemo {
    @SentinelResource(blockHandlerClass = [BlockHandlerClass::class], fallbackClass = [FallbackClass::class])
    fun greeting1(name: String?): String {
        return Greeting.doGreeting(name)
    }

    @SentinelResource(blockHandlerClass = [BlockHandlerClass::class], fallbackClass = [FallbackClass::class])
    fun greeting2(name: String?): String {
        return Greeting.doGreeting(name)
    }

    @SentinelResource(
        blockHandlerClass = [BlockHandlerClass::class, BlockHandlerClass2::class],
        fallbackClass = [FallbackClass::class, FallbackClass2::class]
    )
    fun greeting3(name: String?): String {
        return Greeting.doGreeting(name)
    }
}
